seed/java-sdk/pagination/src/main/java/com/seed/pagination/resources/users/UsersClient.java

```diff
@@ -71,8 +77,18 @@ public ListUsersPaginationResponse listWithCursorPagination(
             }
             Response response = client.newCall(okhttpRequest).execute();
             ResponseBody responseBody = response.body();
+
             if (response.isSuccessful()) {
-                return ObjectMappers.JSON_MAPPER.readValue(responseBody.string(), ListUsersPaginationResponse.class);
+                ListUsersPaginationResponse parsedResponse = ObjectMappers.JSON_MAPPER.readValue(
+                    responseBody.string(), ListUsersPaginationResponse.class);
+                Optional<String> startingAfter = parsedResponse.getPage()
+                    .flatMap(Page::getNext).map(NextPage::getStartingAfter);
+                ListUsersCursorPaginationRequest nextRequest = ListUsersCursorPaginationRequest.builder()
+                    .from(request)
+                    .startingAfter(startingAfter)
+                    .build();
+                return new SyncPagingIterable<>(startingAfter.isPresent(), parsedResponse.getData(),
+                    () -> listWithCursorPagination(nextRequest, requestOptions));
             }
```

Comment:
Example block for offset pagination (from hume, includes handling a response that's an `Optional<List< ... >>`)

```java
            if (response.isSuccessful()) {
                FilePage parsedResponse = ObjectMappers.JSON_MAPPER.readValue(responseBody.string(), FilePage.class);
                FilesListFilesRequest nextRequest = FilesListFilesRequest.builder()
                        .from(request)
                        .pageNumber(
                                request.getPageNumber().map(page -> page + 1).orElse(1))
                        .build();
                return new SyncPagingIterable<>(
                        true, parsedResponse.getContent(), () -> listFiles(nextRequest, requestOptions));
            }
```